docs/details/signal.dox

```diff
@@ -211,32 +211,82 @@ respectively, then the possible batch operations are as follows.
 
 \defgroup signal_func_approx2 approx2
 \ingroup approx_mat
 
-approx2 performs interpolation on data along the first and second dimensions.
-It has three options for the type of interpolation to perform:
-- Nearest neighbor  - \ref AF_INTERP_NEAREST
-- Linear interpolation  - \ref AF_INTERP_LINEAR
-- Bilinear interpolation - \ref AF_INTERP_BILINEAR
-- Cubic interpolation - \ref AF_INTERP_CUBIC
-
-Interpolation is performed assuming input data is equally spaced with indices
-in the range [0, n) along each dimension.
-The positions are sampled with respect to data at these locations.
+approx2 interpolates data along two dimensions. Interpolation computes
+for values of unknown points out of known ones.
+
+The multi-dimensional (>=2D) input array, `in`, contains known values
+to be used for interpolation. These known values lie on a 2D
+interpolation grid - a two-dimensional, uniformly-spaced grid of
+points. Interpolation is performed to obtain values for any unknown
+points on the grid. The index values of the grid can be specified with
+the `grid_beg` and `grid_step` arguments.
+
+Two position arrays (<=2D), `pos0` and `pos1`, contain a combination
+of known and unkown positions/indices on the interpolation grid. Each
+of the position arrays must be of at most two dimensions. The index
+values for each of the position arrays is read along the dimensions
+specified by `pos0_interp_dim` and `pos1_interp_dim`, respectively.
+
+Any unknown position/index values that lie within the interpolation
+grid range will be calculated for using interpolation. Unknown points
+that are outside of the interpolation grid range will be replaced with
+the `off_grid` value. Known points will be replaced with corresponding
+values from the input array.
+
+The two interpolation dimensions, `pos0_interp_dim` and
+`pos1_interp_dim`, correspond to the dimensions that the `pos0` and
+`pos1` arrays will be measured across the interpolation
+grid. Interpolation is thus performed along both specified dimensions
+of the input array.
+
+The uniformly-spaced two-dimensional interpolation grid on which the
+input values lie on can be adjusted with the corresponding grid begin
+and step arguments. A standard uniform interpolation grid with equally
+spaced indices in the range of [0,n) can be achieved with grid begin
+and step values of 0 and 1, respectively. A standard uniform grid will
+be used in the case that the interpolating dimension as well as the
+grid start and step values are not passed in.
+
+All of the interpolation types defined in \ref af_interp_type are
+supported by approx2(). ArrayFire performs linear interpolation by
+default.
+
+Unless specified, linear interpolation is performed by default. Refer
+to \ref af_interp_type for more information about ArrayFire's
+interpolation types.
+
+Interpolation positions outside of the interpolation grid range are
+not extrapolated. Instead, those values default to `off_grid`, whose
+default value is 0.
```

Comment:
Looks like last two paragraphs are giving out only redundant info that is already explained in earlier paragraphs.